tooling/lsp/src/requests/completion.rs

```diff
@@ -381,6 +1149,79 @@ impl<'a> NodeFinder<'a> {
         }
     }
 
+    fn builtin_functions_completion(&mut self, prefix: &str) {
+        if name_matches("assert", prefix) {
+            self.completion_items.push(snippet_completion_item(
+                "assert(…)",
+                CompletionItemKind::FUNCTION,
+                "assert(${1:predicate})",
+                Some("fn(T)".to_string()),
+            ));
+            self.completion_items.push(snippet_completion_item(
+                "assert(…)",
+                CompletionItemKind::FUNCTION,
+                "assert(${1:predicate}, ${2:message})",
+                Some("fn(T, str)".to_string()),
+            ));
+        }
+
+        if name_matches("assert_eq", prefix) {
+            self.completion_items.push(snippet_completion_item(
+                "assert_eq(…)",
+                CompletionItemKind::FUNCTION,
+                "assert_eq(${1:lhs}, ${2:rhs})",
+                Some("fn(T, T)".to_string()),
+            ));
+            self.completion_items.push(snippet_completion_item(
+                "assert_eq(…)",
+                CompletionItemKind::FUNCTION,
+                "assert_eq(${1:lhs}, ${2:rhs}, ${3:message})",
+                Some("fn(T, T, str)".to_string()),
+            ));
+        }
+
+        for keyword in ["false", "true"] {
+            if name_matches(keyword, prefix) {
+                self.completion_items.push(simple_completion_item(
+                    keyword,
+                    CompletionItemKind::KEYWORD,
+                    Some("bool".to_string()),
+                ));
+            }
+        }
+    }
+
+    fn builtin_types_completion(&mut self, prefix: &str) {
+        for typ in [
+            "bool",
+            "i8",
+            "i16",
+            "i32",
```

Comment:
This list looks like it'll be very easy to forget to update when we add more builtin types for comptime code.

Reply:
Good point. I was wondering how we'd dehardcode it... for example some token keywords are type names (FunctionDefinition, etc.) so we could iterate over all keywords and ask whether a keyword corresponds to a built-in type (and which one), so if we add a new keyword for a type we'd have to consider that new case.

We could maybe do the same thing with `Token::IntType`, though the bit sizes are `u32` and not an enum...

Let me know if this approach or a different one would be good and I can work on it.

I guess we'd do something similar with `assert` and `assert_eq` (they come from keywords)

Reply:
I think iterating through the tokens is more or less the only way I can think of to get an error if we miss any. There's a similar test we have for keywords in the tokens file.

Reply:
> There's a similar test we have for keywords in the tokens file.

Oh, nice! I was looking at how I can iterate over all enum values, and found that there's a crate called `strum`, but I wasn't sure about adding that dependency... but we already have it! So, great, I'll try to do it.

Reply:
Done! The only thing that's still hardcoded are the several integer types, and the `false` and `true` keywords. But maybe those won't get larger...